### PR TITLE
Delete channel and Expanded textfield

### PR DESCRIPTION
**Pull request by Lord-Chris**
https://user-images.githubusercontent.com/58702861/135496629-affa59b1-00c5-4a9e-be28-a8983d1eef80.mp4

### CODE REVIEW

File: lib/general_widgets/expandable_textfield.dart [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                    color: AppColors.darkGreyColor,
+                  ),
+                ),
+              ),
+            ),
+          ),
+          Visibility(
+            visible: isExpanded,
+            child: Expanded(
+              child: _buildTextField(),
```
Comment (jeremiahseun) [inline lib/general_widgets/expandable_textfield.dart:60]: Please make this a class widget and not a function widget.
Example is BuildTextField() and not _buildTextField()

File: lib/package/base/server-request/api/zuri_api.dart [hunk 2]
```diff
@@ -102,10 +102,13 @@ class ZuriApi implements Api {
     }
   }
 
-  Future<dynamic> delete(String string) async {
+  Future<dynamic> delete(String string, {String? token}) async {
```
Comment (jeremiahseun) [inline lib/package/base/server-request/api/zuri_api.dart:105]: Have you updated this function in the api.dart?
If not, please go and add what you added here in the api.dart file

File: lib/package/base/server-request/api/zuri_api.dart [hunk 3]
```diff
@@ -322,7 +325,6 @@ class ZuriApi implements Api {
     }
   }
 
-  @override
```
Comment (jeremiahseun) [inline lib/package/base/server-request/api/zuri_api.dart:325]: Please why are you removing the @override?

File: lib/package/base/server-request/api/zuri_api.dart [hunk 4]
```diff
@@ -390,7 +392,6 @@ class ZuriApi implements Api {
   }
 
   // THIS SERVICE IS FOR JOINED ROOMS FOR ACTIVE DMs
-  @override
```
Comment (jeremiahseun) [inline lib/package/base/server-request/api/zuri_api.dart:393]: With this too

File: lib/ui/view/channel/channel_info/widgets/sixth_section.dart [hunk 5]
```diff
@@ -1,54 +1,63 @@
...
-            crossAxisAlignment: CrossAxisAlignment.start,
-            children: [
-              Text(
-                'Archive Channel',
-                style: AppTextStyles.archiveTextStyle2.copyWith(fontSize: 14.sp),
+  Widget build(BuildContext context, ChannelInfoViewModel model) {
+    return GestureDetector(
+      onTap: () => model.deleteChannel(channelDetail),
+      child: Container(
+        height: 136.h,
```
Comment (jeremiahseun) [inline lib/ui/view/channel/channel_info/widgets/sixth_section.dart:19]: Will it scale?

File: lib/ui/view/channel/channel_info/widgets/sixth_section.dart [hunk 6]
```diff
@@ -1,54 +1,63 @@
...
+        width: 395.w,
+        alignment: Alignment.center,
+        padding: EdgeInsets.fromLTRB(16.37.w, 18.h, 0, 0),
+        decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(6.r),
+            border: Border.all(width: 1.w, color: AppColors.borderColor),
+            boxShadow: [
+              BoxShadow(
+                color: Colors.grey.withOpacity(0.5),
+                spreadRadius: 5.r,
```
Comment (jeremiahseun) [inline lib/ui/view/channel/channel_info/widgets/sixth_section.dart:29]: And this?

File: lib/ui/view/threads/thread_detail/thread_detail_view.dart [hunk 7]
```diff
@@ -22,218 +21,103 @@ class ThreadDetailView extends StatelessWidget {
     final _messageController = TextEditingController();
```
Comment (jeremiahseun) [inline lib/ui/view/threads/thread_detail/thread_detail_view.dart:21]: This should not be here. Generate it that the stateless widget will go **with $ThreadDetailView**.
That file will give you the text controller that you need.

Achieving this is by making the view model of this page, **FormViewModel**.

**Review by jeremiahseun — DISMISSED** [review]
Nice work. Please make changes

File: lib/package/base/server-request/api/zuri_api.dart (hunk 3 above)
Comment (Lord-Chris) [reply lib/package/base/server-request/api/zuri_api.dart:325]: Its not in the interface so there's nothing for it to override

File: lib/general_widgets/expandable_textfield.dart (hunk 1 above)
Comment (Lord-Chris) [reply lib/general_widgets/expandable_textfield.dart]: Solved

File: lib/package/base/server-request/api/zuri_api.dart (hunk 2 above)
Comment (Lord-Chris) [reply lib/package/base/server-request/api/zuri_api.dart]: Done

File: lib/package/base/server-request/api/zuri_api.dart (hunk 4 above)
Comment (Lord-Chris) [reply lib/package/base/server-request/api/zuri_api.dart]: Same as above

File: lib/ui/view/channel/channel_info/widgets/sixth_section.dart (hunk 5 above)
Comment (Lord-Chris) [reply lib/ui/view/channel/channel_info/widgets/sixth_section.dart]: It has been removed

File: lib/ui/view/channel/channel_info/widgets/sixth_section.dart (hunk 6 above)
Comment (Lord-Chris) [reply lib/ui/view/channel/channel_info/widgets/sixth_section.dart]: Same here

File: lib/ui/view/threads/thread_detail/thread_detail_view.dart (hunk 7 above)
Comment (Lord-Chris) [reply lib/ui/view/threads/thread_detail/thread_detail_view.dart]: It doesn't require a form view model, I changed it to use hook widget's method

**Comment by pauleke65** [comment]
@Lord-Chris Awesome work.
Please resolve conflicts so that your PR can be merged

**Comment by Lord-Chris** [comment]
Conflicts resolved

**Review by pauleke65 — APPROVED** [review]
LG

**Comment by pauleke65** [comment]
@jeremiahseun

**Review by YenumJay — APPROVED** [review]
Great Job, Looks good.

File: lib/package/base/server-request/api/zuri_api.dart (hunk 4 above)
Comment (Lord-Chris) [reply lib/package/base/server-request/api/zuri_api.dart]: Same with above, nothing for it to override

**Comment by pauleke65** [comment]
@jeremiahseun Requested conversations resolved, will be dismissing review and merging PR now

**Review by Lord-Chris — COMMENTED** [review]
It is great

**Pull request merged**